### PR TITLE
bench: add send_tps metric

### PR DESCRIPTION
**Pull request by grooviegermanikus**
This PR adds _send_tps_ to bench metrics. The value is calculated from the **gross time spent calling RPC endpoint**.

**Log**
> 2023-10-04T13:10:42.046142Z  INFO bench: Run 1: Sent and Confirmed 30 tx(s) in Metric { txs_sent: 30, txs_confirmed: 30, txs_un_confirmed: 0, average_confirmation_time_ms: 1592.2333333333333, average_time_to_send_txs: 1.1666666666666667, average_transaction_bytes: 1186.0, send_tps: 279.85596746207955, total_sent_time: 35.698375ms, total_transaction_bytes: 35580, total_confirmation_time: 47.767245668s, total_gross_send_time_ms: 107.198 } with    
2023-10-04T13:10:42.046217Z  INFO bench: Run 2: Sent and Confirmed 30 tx(s) in Metric { txs_sent: 30, txs_confirmed: 30, txs_un_confirmed: 0, average_confirmation_time_ms: 1529.9666666666667, average_time_to_send_txs: 1.2666666666666666, average_transaction_bytes: 1186.0, send_tps: 270.79233838210604, total_sent_time: 38.520332ms, total_transaction_bytes: 35580, total_confirmation_time: 45.899511625s, total_gross_send_time_ms: 110.786 } with    
2023-10-04T13:10:42.046230Z  INFO bench: Run 3: Sent and Confirmed 30 tx(s) in Metric { txs_sent: 30, txs_confirmed: 30, txs_un_confirmed: 0, average_confirmation_time_ms: 2007.3, average_time_to_send_txs: 1.4, average_transaction_bytes: 1186.0, send_tps: 258.5649644473174, total_sent_time: 42.890165ms, total_transaction_bytes: 35580, total_confirmation_time: 60.219622169s, total_gross_send_time_ms: 116.025 } with    
2023-10-04T13:10:42.046331Z  INFO bench: Avg Metric Metric { txs_sent: 30, txs_confirmed: 30, txs_un_confirmed: 0, average_confirmation_time_ms: 1709.8333333333333, average_time_to_send_txs: 1.3, average_transaction_bytes: 1186.0, send_tps: 269.4538171127125, total_sent_time: 39.036ms, total_transaction_bytes: 35580, total_confirmation_time: 51.295459s, total_gross_send_time_ms: 111.33633333333334 }

**metrics.csv**
> txs_sent,txs_confirmed,txs_un_confirmed,average_confirmation_time_ms,average_time_to_send_txs,average_transaction_bytes,send_tps
30,30,0,1592.2333333333333,1.1666666666666667,1186.0,279.85596746207955
30,30,0,1529.9666666666667,1.2666666666666666,1186.0,270.79233838210604
30,30,0,2007.3,1.4,1186.0,258.5649644473174
30,30,0,1709.8333333333333,1.3,1186.0,269.4538171127125

### CODE REVIEW

File: bench/src/main.rs [hunk 1]
```diff
@@ -131,7 +129,7 @@ async fn main() {
 
     let avg_metric = Metric::from(avg_metric);
 
-    info!("Avg Metric {avg_metric:?}",);
+    info!("Avg Metric {:?}", avg_metric);
```
Comment (aniketfuryrocks) [inline bench/src/main.rs:132]: wondering what's wrong with the old style 🤔

File: bench/src/metrics.rs [hunk 2]
```diff
@@ -44,10 +47,20 @@ impl Metric {
                 self.total_sent_time.as_millis() as f64 / self.txs_sent as f64;
         }
 
+        if self.total_gross_send_time_ms > 0.01 {
+            self.send_tps =
+                1000.0 * self.txs_sent as f64 / self.total_gross_send_time_ms;
```
Comment (aniketfuryrocks) [inline bench/src/metrics.rs:52]: please use () to make this more readable and obvious

File: bench/src/main.rs (hunk 1 above)
Comment (grooviegermanikus) [reply bench/src/main.rs:132]: did not know it is old ;) my IDE does not offer the same commands. I always used the latter style? why are you advocating for the former?

File: bench/src/metrics.rs (hunk 2 above)
Comment (grooviegermanikus) [reply bench/src/metrics.rs:52]: very good point! will do

File: bench/src/main.rs (hunk 1 above)
Comment (grooviegermanikus) [reply bench/src/main.rs]: reverted ..

File: bench/src/main.rs [hunk 3]
```diff
@@ -167,11 +168,13 @@ async fn bench(
     transaction_size: TransactionSize,
 ) -> Metric {
     let map_of_txs: Arc<DashMap<Signature, TxSendData>> = Arc::new(DashMap::new());
+    let total_gross_send_time = Arc::new(OnceCell::new());
```
Comment (godmodegalactus) [inline bench/src/main.rs:171]: Return the instance instead of using Onecell.

**Review by godmodegalactus — CHANGES_REQUESTED** [review]

File: bench/src/main.rs (hunk 3 above)
Comment (grooviegermanikus) [reply bench/src/main.rs:171]: how that? can you show me?

**Review by godmodegalactus — DISMISSED** [review]

File: bench/src/main.rs [hunk 4]
```diff
@@ -154,6 +154,10 @@ struct TxSendData {
     transaction_bytes: u64,
 }
 
+struct ApiCallerResult {
```
Comment (godmodegalactus) [inline bench/src/main.rs:157]: Dont know why you created a strucutre for one element.

**Review by godmodegalactus — APPROVED** [review]

File: bench/src/main.rs (hunk 4 above)
Comment (grooviegermanikus) [reply bench/src/main.rs:157]: returning only one specific and unrelated value from the huge task feels very wrong. i.e. the task is **not** about getting the _bench_start_time_.

**Pull request merged**